src/CgStr.hpp

```diff
@@ -16,21 +16,21 @@ class CgStr {
         : m_data(std::move(s))
     { }
 
-    CgStr(CgStr&& other)
+    CgStr(CgStr&& other) noexcept
         : m_data(std::move(other.m_data))
     {
         other.m_data.data = nullptr;
     }
 
-    CgStr& operator=(CgStr&& other) {
+    CgStr& operator=(CgStr&& other) noexcept {
         destroy();
         m_data = std::move(other.m_data);
         other.m_data.data = nullptr; // HACK Undocumented behavior.
         assert(!other.valid());
         return *this;
     }
 
-    ~CgStr() {
+    ~CgStr() noexcept {
```

Comment:
I prefer to rely on C++11's default `noexcept` for destructors, so that if I ever have a `noexcept(false)` dtor, it sticks out even more. So please remove that one.